sdk/core/azure-core/src/main/java/com/azure/core/exception/HttpResponseException.java

```diff
@@ -69,6 +69,19 @@ public HttpResponseException(final String message, final HttpResponse response,
         this.response = response;
     }
 
+    /**
+     * Initializes a new instance of the HttpResponseException class.
+     *
+     * @param message The exception message.
+     * @param response The {@link HttpResponse} received that is associated to the exception.
+     * @param error The {@link HttpResponseError} associated with this exception.
+     */
+    public HttpResponseException(final String message, final HttpResponse response, final HttpResponseError error) {
```

Comment:
Doesn't this result in an ambiguous call-site as the `(String, HttpResponse, Object)` overload could also accept `HttpResponseError`. Have we looking into just using that instead?

Reply:
If you are referring to calls like `("string", response, null)` that cause ambiguity then there's already another overload that has a similar shape - [public HttpResponseException(String message, HttpResponse response, Throwable cause)](https://apiview.dev/Assemblies/Review/ce1fd1f49fae4aa7bffc859881c6af0a?diffRevisionId=26e416a467e74ce683da912fb13ee5c6&doc=False&diffOnly=True#com.azure.core.exception.HttpResponseException). So, callers using `null` as 3rd arg should already be casting it to either Object or Throwable explicitly.